src/backend/distributed/commands/multi_copy.c

```diff
@@ -465,7 +465,7 @@ CopyToExistingShards(CopyStmt *copyStatement, char *completionTag)
 	 * PG_CATCH. Otherwise, it may be undefined in the PG_CATCH (see sigsetjmp
 	 * documentation).
 	 */
-	shardConnectionHash = CreateShardConnectionHash();
+	copyConnectionHash = CreateShardConnectionHash(MessageContext);
```

Comment:
Eh, feel like saying that that's out of scope, as right now `copy` manually commits its transactions immediately rather than using a transaction event callback. The hash change is necessary to have multi-DDL transactions targeting multiple distributed relations; extending it to `copy` would require changes unrelated to enabling multi-DDL, and I think we'll be changing enough of this stuff in short order anyhow.